cmd/dhtctl/main.go

```diff
@@ -11,67 +11,60 @@ import (
 	"github.com/optmzr/d7024e-dht/store"
 )
 
-func put(c *rpc.Client, val string) {
+func put(c *rpc.Client, value string) {
 	put := ctl.Put{
-		Val: val,
+		Value: value,
 	}
-	var reply bool
+	var key store.Key
 
 	// The RPC call
-	err := c.Call("API.Put", put, &reply)
+	err := c.Call("API.Put", put, &key)
 	if err != nil {
 		log.Fatal("Put error:", err)
 	}
 
-	// Debug (reply)
-	if reply {
-		fmt.Println("RPC successful")
-	}
+	log.Printf("Hash: %v\n", key)
 }
 
 func get(c *rpc.Client, key store.Key) {
 	get := ctl.Get{
 		Key: key,
 	}
-	var reply bool
+	var value string
 
 	// The RPC call
-	err := c.Call("API.Get", get, &reply)
+	err := c.Call("API.Get", get, &value)
 	if err != nil {
-		log.Fatal("Get error:", err)
+		log.Fatalln("Get error:", err)
 	}
 
-	// Debug (reply)
-	if reply {
-		fmt.Println("RPC successful")
-	}
+	log.Printf("Value: %s\n", value)
 }
 
 func ping(c *rpc.Client, id node.ID) {
 	ping := ctl.Ping{NodeID: id}
-	var reply []byte
+	var challenge []byte
 
 	// The RPC call
-	err := c.Call("API.Ping", ping, &reply)
+	err := c.Call("API.Ping", ping, &challenge)
 	if err != nil {
-		log.Fatal("Ping error:", err)
+		log.Fatalln("Ping error:", err)
 	}
 
-	fmt.Println("Ping response: ", reply)
+	fmt.Printf("Ping response: %x\n", challenge)
 }
 
-func exit(c *rpc.Client, id node.ID) {
-	var reply bool
+func exit(c *rpc.Client) {
```

Comment:
Should'n we take a nodeID as an argument? What if we want to terminate another node?

Reply:
The lab spec. says "the node", so I thought that we'd skip that feature :)